src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -103,18 +125,184 @@ def validate(self, node1, node2, options=None):
             if not (SymbolicMaths.equal(node1.start_expr, node2.start_expr) and
                     SymbolicMaths.equal(node1.stop_expr, node2.stop_expr) and
                     SymbolicMaths.equal(node1.step_expr, node2.step_expr)):
+                # TODO #257: This transformation assumes that all domain loop
+                # bodies have only POINTWISE accesses to fields and does not
+                # perform any dependency analysis.
+                # This is wrong and it will generate incorrect code for any
+                # kernel with STENCIL access.
                 raise TransformationError(LazyString(
                     lambda node1=node1, node2=node2:
                         f"Error in {self.name} transformation. Loops do not "
                         f"have the same iteration space:\n"
-                        f"{node1.view()}\n"
-                        f"{node2.view()}"))
+                        f"{node1.debug_string()}\n"
+                        f"{node2.debug_string()}"))
+
+            vars1 = VariablesAccessInfo(node1)
```

Comment:
Doesn't this need to be outdented by one level? All these tests are skipped if the loop is a PSyloop. While this might (??) imply that the variable names are identical (since it's code created by PSyclone), this will also skip the variables tests at around line 167.

Reply:
@sergisiso said to do this, I'm not sure

Reply:
Since no tests fail I think its probably fine to have this at this level on indentation?

Reply:
The indentation is wrong ... BUT: if you try to fix this (and actually execute this code with e.g. an gocean example). you will get a crash in the dependency tools, since it turns out in case of a kernel we get incorrect data types (I am debugging this atm, string instead of PSyIR nodes). Add a TODO #257, and I am putting some details into that ticket.

Reply:
I am aware this needs to be outdented at some point. Ideally we want to even be able to fuse domain and generic loops. But this was out of scope for this PR, we just want to maintain the same functionality we had in the nemo loop but without the specialized transformation. I asked @LonelyCat124 to add a note at the transformation docstring about the limitations but we can also add a TODO comment here.

Reply:
I've just submitted #2497, which replaces the string in the access information in case of a GOcean kernel with proper PSyIR expressions, so that the partition call (used here) does not fail anymore if it would be outdented.

There is no need to change anything here for now, I am happy to fix this once this PR is merged as part of my PR (I am still fixing CI, so I think you will be in way before me) - it would just be great if you could ping me when merging this PR in case that I miss it :)  Thanks for your work on fusion!